crates/wasmtime/src/runtime/gc/enabled/anyref.rs

```diff
@@ -349,6 +368,72 @@ impl AnyRef {
     pub fn unwrap_i31(&self, store: impl AsContext) -> Result<I31> {
         Ok(self.as_i31(store)?.expect("AnyRef::unwrap_i31 on non-i31"))
     }
+
+    /// Is this `anyref` a `structref`?
+    ///
+    /// # Errors
+    ///
+    /// Return an error if this reference has been unrooted.
+    ///
+    /// # Panics
+    ///
+    /// Panics if this reference is associated with a different store.
+    pub fn is_struct(&self, store: impl AsContext) -> Result<bool> {
+        self._is_struct(store.as_context().0)
+    }
+
+    pub(crate) fn _is_struct(&self, store: &StoreOpaque) -> Result<bool> {
+        // NB: Can't use `AutoAssertNoGc` here because we only have a shared
+        // context, not a mutable context.
```

Comment:
Performing a GC requires `&mut` though, right? So given `&StoreOpaque` that's static proof that a GC won't happen? 

Musing though: because `unchecked_try_gc_ref` returns a lifetime bound to `&StoreOpaque` doesn't that mean that GC is already statically disallowed while we're working with the `gc_ref`? If that's the case could the `unchecked_` prefix be removed? (fine to defer this to a future PR of course)

Reply:
Spun this out to https://github.com/bytecodealliance/wasmtime/issues/8940